exchange/bidder.go

```diff
@@ -751,3 +744,40 @@ func hasShorterDurationThanTmax(ctx bidderTmaxContext, tmaxAdjustments TmaxAdjus
 	}
 	return false
 }
+
+func setRequestBody(req *adapters.RequestData, endpointCompression string) ([]byte, error) {
+	var requestBody []byte
+
+	switch strings.ToUpper(endpointCompression) {
+	case Gzip:
+		// Compress to GZIP
+		var b bytes.Buffer
+		w := gzipWriterPool.Get().(*gzip.Writer)
+		w.Reset(&b)
+		_, err := w.Write([]byte(requestBody))
+		if err != nil {
+			return nil, err
+		}
+		err = w.Close()
+		if err != nil {
+			return nil, err
+		}
+		requestBody = b.Bytes()
```

Comment:
Instead of allocating a new byte slice, could we return `b`? That would also save us from allocating a new byte buffer to attach the body to the request. Just be careful to reset the writer to `nil` or `io.Discard` before returning it to the pool to prevent leaking the buffer reference.

Reply:
Nevermind. The extra call to `Reset` is very expensive.

Reply:
Also confirmed the memory savings is minimal as a byte buffer is a small object.